### PR TITLE
[TC-8978] Add shipment polling

### PR DESCRIPTION
**Pull request by marcmatt**
## Description

https://tradecloud.atlassian.net/browse/TC-8978

### Scope
This PR adds shipment polling to the API manial

Please review https://app.gitbook.com/o/-LNyIUZSvpZWZ81yEr7V/s/-LNyJ9UuwLCjTvA2sqQR-887967055/~/revisions/MifRCHBvioSaeHd6eVnH/introduction/api/webhook-vs-polling

### Submitter pre-flight check

- [x] Review your documentation
- [x] Add labels `needs reviewing`
- [x] Assign PR and JIRA ticket to 'Reviewer' (in 'Reviewing' state)

### Reviewer pre-flight check

- [ ] Review documentation in PR
- [ ] Remove label `needs reviewing`
- [ ] Merge the PR and remove the release for this branch from Gitbook
- [ ] Set Jira ticket to 'Released'

### CODE REVIEW

File: api/webhook-vs-polling.md [hunk 1]
```diff
@@ -93,20 +92,20 @@ Con's:
...
 ### Polling usage
 
-#### Step 1. Fetch updated orders periodically
+#### Step 1. Fetch updated orders or shipments periodically
 
-Fetch every polling period, typically 5 minutes, all orders which are new or changed since last date time.
+Fetch every polling period, typically 5 minutes, all orders or shipments which are new or changed since last date time.
 
 * Use the latest `lastUpdatedAt` from previous poll request in the `lastUpdatedSince` filter.
 * Sorting is set automatically to `lastUpdatedAt` order `asc` \(latest `lastUpdatedAt` will be in the last order in the response body\)
```
Comment (Sjorstradecloud) [inline api/webhook-vs-polling.md:106]: I think the sentence below fitst better. 
(latest `lastUpdatedAt` will be last in the response body\

File: api/webhook-vs-polling.md [hunk 2]
```diff
@@ -146,18 +145,26 @@ application/json
...
+See the [Search orders OpenAPI Specification](https://swagger-ui.accp.tradecloud1.com/?url=https://api.accp.tradecloud1.com/v2/order-search/specs.yaml#/order-search/searchRoute).
+
+* Use the `lastUpdatedAt` on order line level to filter on if the line has been changed.
+* Use the `status` field to filter on order process and logistics status.
+
+See the [Search shipments OpenAPI Specification](https://swagger-ui.accp.tradecloud1.com/?url=https://api.accp.tradecloud1.com/v2/shipment/specs.yaml#/shipment/searchShipmentsRoute).
 
-* Use the `lastUpdatedAt` on order line level to filter on the line has been changed.
-* Use the `status` field to filter on process and logistics status.
+* Use the `status` field to filter on shipment process and logistics status.
```
Comment (Sjorstradecloud) [inline api/webhook-vs-polling.md:161]: This is not yet implemented

File: api/webhook-vs-polling.md [hunk 3]
```diff
@@ -146,18 +145,26 @@ application/json
...
 
 #### Step 4. Store the `lastUpdatedAt` for the next polling request
```
Comment (Sjorstradecloud) [inline api/webhook-vs-polling.md:163]: Step 3 is missing.

**Review by Sjorstradecloud — COMMENTED** [review]
Left some small comments

File: api/webhook-vs-polling.md [hunk 4]
```diff
@@ -28,21 +27,22 @@ When using `POST` or `PUT` the webhook request body will contain:
...
 * `orderEvent`: The actual order event, see [OrderEvent](https://swagger-ui.accp.tradecloud1.com/?url=https://api.accp.tradecloud1.com/v2/order-webhook-client/specs.yaml#/order-webhook%20endpoints/webhookPost) \(click "Model" and "OrderEvent"\) and [Receive order response](./).
-* `orderDocumentsEvent`: Or the actual order documents event, see see [OrderDocumentEvent](https://swagger-ui.accp.tradecloud1.com/?url=https://api.accp.tradecloud1.com/v2/order-webhook-client/specs.yaml#/order-webhook%20endpoints/webhookPost) \(click "Model" and "OrderDocumentsEvent"\).
+* `orderDocumentsEvent`: Or the actual order documents event, see [OrderDocumentEvent](https://swagger-ui.accp.tradecloud1.com/?url=https://api.accp.tradecloud1.com/v2/order-webhook-client/specs.yaml#/order-webhook%20endpoints/webhookPost) \(click "Model" and "OrderDocumentsEvent"\).
+* `shipmentEvent`: The actual shipment, see [Shipment](https://swagger-ui.accp.tradecloud1.com/?url=https://api.accp.tradecloud1.com/v2/shipment-webhook-connector/specs.yaml#/shipment-webhook%20endpoints/webhookPost) \(click "Model" and "Shipment"\).
```
Comment (roy-tc) [inline api/webhook-vs-polling.md:31]: "Or the actual shipment".
perhaps make this a sublist, to indicate that you will either get one of these 3 events, but never multiple at the same time.

File: api/webhook-vs-polling.md [hunk 5]
```diff
@@ -28,21 +27,22 @@ When using `POST` or `PUT` the webhook request body will contain:
...
 
 Use `POST` or `PUT` when:
```
Comment (roy-tc) [inline api/webhook-vs-polling.md:33]:
```suggestion
Use `POST` when:
```

PUT is not supported by the `order-webhook-connector` or `shipment-webhook-connector`

File: api/webhook-vs-polling.md [hunk 6]
```diff
@@ -28,21 +27,22 @@ When using `POST` or `PUT` the webhook request body will contain:
...
 
-* You want to receive real time order events.
-* You want to receive the order event content.
-* You only want to receive selected order events.
+* You want to receive real time order or shipment events.
+* You want to receive the order event or shipment event content.
+* You only want to receive selected order or shipment events.
```
Comment (roy-tc) [inline api/webhook-vs-polling.md:37]:
```suggestion
* You only want to receive order or shipment events of a specific type.
```

File: api/webhook-vs-polling.md [hunk 7]
```diff
@@ -64,25 +64,24 @@ Use `GET` when:
 {% hint style="info" %}
 Pro's:
 
-* Semi real time, receive the order within seconds.
-* You can filter on which order events to receive \(in the company integration settings only\).
+* Semi real time, receive the order or shipment within seconds.
```
Comment (roy-tc) [inline api/webhook-vs-polling.md:67]:
```suggestion
* Real time, receive the order or shipment within a second.
```

This isn't any slower than `POST`, right?

File: api/webhook-vs-polling.md [hunk 8]
```diff
@@ -64,25 +64,24 @@ Use `GET` when:
...
 * You need to obtain and configure a public SSL certificate.
 {% endhint %}
 
 ## The polling pattern
 
-Check if there are new or updated orders every polling period, typically 5 minutes, by using the last updated date time stamp of the last fetched order.
+Check if there are new or updated orders or shipments every polling period, typically 5 minutes, by using the last updated date time stamp of the last fetched order or shipment.
 
-The polling pattern is most suitable for companies with low volume order responses and not willing to invest in a web server, firewall and SSL certificate.
+The polling pattern is most suitable for companies with low volume orders, and not willing to invest in a web server, firewall and SSL certificate.
```
Comment (roy-tc) [inline api/webhook-vs-polling.md:84]: Nit: somehow I'm reading this quite negative "not willing to invest in..."

File: api/webhook-vs-polling.md [hunk 9]
```diff
@@ -93,20 +92,20 @@ Con's:
 
 * Not real time, a polling period is typically 5 mins.
 * You need to build or configure a periodic polling pattern at your side.
-* You cannot filter on which order events to receive, you will receive any order line change.
-* You cannot see what order event happened.
+* You cannot filter on which order or shipments events to receive, you will receive any order line change.
```
Comment (roy-tc) [inline api/webhook-vs-polling.md:95]:
```suggestion
* You cannot filter on which order or shipments events to act on, you will receive any order line / shipment line change.
```

File: api/webhook-vs-polling.md [hunk 10]
```diff
@@ -146,18 +145,26 @@ application/json
...
+{% hint style="info" %}
+[Search shipments OpenAPI Specification](https://swagger-ui.accp.tradecloud1.com/?url=https://api.accp.tradecloud1.com/v2/shipment/specs.yaml#/shipment/searchShipmentsRoute)
+{% endhint %}
+
 #### Step 2. Process the orders in the search response body
 
-See the [Search orders OpenAPI Specification](https://swagger-ui.accp.tradecloud1.com/?url=https://api.accp.tradecloud1.com/v2/order-search/specs.yaml#/order-search/searchRoute) and [Receive order response](./) for order fields descriptions.
+See the [Search orders OpenAPI Specification](https://swagger-ui.accp.tradecloud1.com/?url=https://api.accp.tradecloud1.com/v2/order-search/specs.yaml#/order-search/searchRoute).
+
+* Use the `lastUpdatedAt` on order line level to filter on if the line has been changed.
```
Comment (roy-tc) [inline api/webhook-vs-polling.md:156]:
```suggestion
* Use the `lastUpdatedAt` on order line level to filter the lines that have been changed.
```

File: api/webhook-vs-polling.md [hunk 11]
```diff
@@ -146,18 +145,26 @@ application/json
...
+* Use the `status` field to filter on order process and logistics status.
+
+See the [Search shipments OpenAPI Specification](https://swagger-ui.accp.tradecloud1.com/?url=https://api.accp.tradecloud1.com/v2/shipment/specs.yaml#/shipment/searchShipmentsRoute).
```
Comment (roy-tc) [inline api/webhook-vs-polling.md:159]: Should this be 2 separate blocks? It feels a bit weird.
Perhaps link both API specs, and then provide intructions (instructions are the same, right?)

File: order/supplier/receive/README.md [hunk 12]
```diff
@@ -24,8 +24,7 @@ First choose either the webhook API or the polling API to receive orders:
 * `version`: the  Tradecloud order version number.
 * `eventDates`: some key order event date/times.
 * `meta`: meta information, including source and trace info, about this messsage
-* `lastUpdatedAt`: is the latest date time the order has been changed, useful for polling.
-
```
Comment (roy-tc) [inline order/supplier/receive/README.md:28]: Nit; newline should remain here

**Review by roy-tc — COMMENTED** [review]
Looks good! Left some suggestions

File: api/webhook-vs-polling.md (hunk 8 above)
Comment (marcmatt) [reply api/webhook-vs-polling.md:84]: That is what I mean, Hollandse Zuinigheid

File: api/webhook-vs-polling.md (hunk 2 above)
Comment (marcmatt) [reply api/webhook-vs-polling.md:161]: Removed

File: api/webhook-vs-polling.md (hunk 3 above)
Comment (marcmatt) [reply api/webhook-vs-polling.md:163]: Renamed to `3`, not sure where `3` went

File: api/webhook-vs-polling.md (hunk 4 above)
Comment (marcmatt) [reply api/webhook-vs-polling.md:31]: Created two lists

File: api/webhook-vs-polling.md (hunk 5 above)
Comment (marcmatt) [reply api/webhook-vs-polling.md]: Removed PUT everywhere

File: api/webhook-vs-polling.md (hunk 9 above)
Comment (marcmatt) [reply api/webhook-vs-polling.md:95]: Used `any order or shipment change` as lines are not relevant.

File: api/webhook-vs-polling.md (hunk 11 above)
Comment (marcmatt) [reply api/webhook-vs-polling.md:159]: Removed the embedded, duplicate API spec, it was something from the past, we now have real API specs

**Comment by marcmatt** [comment]
@roy-tc @Sjorstradecloud Please review https://app.gitbook.com/o/-LNyIUZSvpZWZ81yEr7V/s/-LNyJ9UuwLCjTvA2sqQR-887967055/~/revisions/MifRCHBvioSaeHd6eVnH/introduction/api/webhook-vs-polling

File: api/webhook-vs-polling.md (hunk 1 above)
Comment (marcmatt) [reply api/webhook-vs-polling.md:110]: Ah, I misread the `order` part. But `latest lastUpdatedAt will be` reallly `in the last order in the response body`
What means `will be last in the response body`?

**Review by roy-tc — APPROVED** [review]
LGTM

File: api/webhook-vs-polling.md (hunk 1 above)
Comment (marcmatt) [reply api/webhook-vs-polling.md:110]: improved

**Pull request merged**